hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/client/ConfiguredRMFailoverProxyProvider.java

```diff
@@ -119,4 +122,31 @@ public synchronized void close() throws IOException {
       }
     }
   }
+
+  /**
+   * YARN Federation mode, the Router is considered as an RM for the client.
+   * We want the client to be able to randomly
+   * Select a Router and support failover when selecting a Router.
+   * The original code always started trying from the first
+   * Router when the client selected a Router,
+   * but this method will support random Router selection.
+   *
+   * For clusters that have not enabled Federation mode, the behavior remains unchanged.
+   *
+   * @param conf Configuration.
+   * @return rmIds
+   */
+  private Collection<String> getRandomOrder(Configuration conf) {
```

Comment:
The name of function is not correct.
It only returns random if the flag is set.

Reply:
I renamed the method for better understanding.